fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java

```diff
@@ -1357,8 +1357,8 @@ private void computeFragmentExecParams() throws Exception {
 
                 // when left table is empty, it's bucketset is empty.
                 // set right table destination address to the address of left table
-                if (destParams.instanceExecParams.size() == 1
-                        && destParams.instanceExecParams.get(0).bucketSeqSet.isEmpty()) {
+                if (destParams.instanceExecParams.size() == 1 && (bucketNum == 0
+                        || destParams.instanceExecParams.get(0).bucketSeqSet.isEmpty())) {
```

Comment:
why add back `destParams.instanceExecParams.get(0).bucketSeqSet.isEmpty()`?

Reply:
for colocate join, the bucketNum is the bucket number in default distribution info. So even the table is empty, the bucketNum is not 0. So add it back to handle the colocate join case